sensor/kubernetes/listener/resources/endpoints.go

```diff
@@ -240,12 +243,47 @@ func (m *endpointManagerImpl) OnDeploymentCreateOrUpdateByID(id string) {
 }
 
 func (m *endpointManagerImpl) onDeploymentCreateOrUpdate(deployment *deploymentWrap) {
+	data := m.endpointDataForDeployment(deployment)
 	updates := map[string]*clusterentities.EntityData{
-		deployment.GetId(): m.endpointDataForDeployment(deployment),
+		deployment.GetId(): data,
+	}
+	if deployment.GetName() == "sensor" {
+		m.updateHeritageData(data, true)
 	}
 	m.entityStore.Apply(updates, false, "OnDeploymentCreateOrUpdateByID")
 }
 
+func (m *endpointManagerImpl) updateHeritageData(data *clusterentities.EntityData, force bool) {
```

Comment:
Not clear why `force` is needed.

Reply:
Yeah, good point. I know why it was needed in the past, but it is not needed anymore. I will remove it.

Reply:
Addressed in [1c15026](https://github.com/stackrox/stackrox/pull/15176/commits/1c1502649638f94077b5b9932edacc050a9f865a)